### ISSUE TITLE
[Phase 5] Experiment Pipeline and Hyperparameter Tuning

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/particle-collision-classifier#6]
## Objective

Create end-to-end experiment pipeline with configuration management and hyperparameter optimization.

**Part of:** #1

## Dependencies

- #2 (Data Infrastructure) complete
- #3 (Feature Engineering) complete
- #4 (Model Implementation) complete
- #5 (Evaluation Framework) complete

### Experiment Runner
- [ ] Create `src/pipeline/runner.py` with ExperimentRunner class
- [ ] Wire together data loading, preprocessing, training, evaluation
- [ ] Support configuration via YAML files

### Hyperparameter Tuner
- [ ] Create `src/pipeline/tuner.py` with grid/random search
- [ ] Implement cross-validation for robust evaluation
- [ ] Track best parameters and scores

### CLI Entry Point
- [ ] Create `scripts/run_experiment.py` with argparse
- [ ] Support dataset selection, model type, sample size
- [ ] Output results to console and file

### Experiment Configuration
- [ ] Create `configs/experiment.yaml` with default settings

### Experiment Runner

```python
# src/pipeline/runner.py
import yaml
from src.data.loader import HEPDataLoader
from src.features.preprocessor import Preprocessor
from src.models.classifier import ParticleClassifier
from src.evaluation.metrics import MetricsCalculator

class ExperimentRunner:
    def __init__(self, config_path=None, **kwargs):
        if config_path:
            with open(config_path) as f:
                self.config = yaml.safe_load(f)
        else:
            self.config = kwargs

    def run(self):
        # Load data
        loader = HEPDataLoader(
            self.config['dataset'],
            sample_size=self.config.get('sample_size', 100000)
        )
        X_train, X_val, X_test, y_train, y_val, y_test = loader.load_split()

        # Preprocess
        preprocessor = Preprocessor()
        X_train_scaled = preprocessor.fit_transform(X_train)
        X_val_scaled = preprocessor.transform(X_val)
        X_test_scaled = preprocessor.transform(X_test)

        # Train
        clf = ParticleClassifier(self.config.get('model_type', 'xgboost'))
        clf.fit(X_train_scaled, y_train,
                eval_set=[(X_val_scaled, y_val)],
                early_stopping_rounds=10)

        # Evaluate
        y_proba = clf.predict_proba(X_test_scaled)
        y_pred = clf.predict(X_test_scaled)

        metrics = MetricsCalculator(y_test, y_pred, y_proba)
        results = metrics.all_metrics()

        return results
```

### CLI Script

```python
# scripts/run_experiment.py
import argparse
from src.pipeline.runner import ExperimentRunner

def main():
    parser = argparse.ArgumentParser()
    parser.add_argument('--config', type=str, default='configs/experiment.yaml')
    parser.add_argument('--dataset', type=str, choices=['higgs', 'hepmass', 'susy'])
    parser.add_argument('--sample-size', type=int, default=100000)
    parser.add_argument('--model', type=str, default='xgboost')
    args = parser.parse_args()

    runner = ExperimentRunner(
        config_path=args.config if not args.dataset else None,
        dataset=args.dataset,
        sample_size=args.sample_size,
        model_type=args.model
    )

    results = runner.run()
    print(f"Results: {results}")

if __name__ == '__main__':
    main()
```

### Experiment Config

```yaml
# configs/experiment.yaml
dataset: higgs
sample_size: 100000
model_type: xgboost
random_state: 42

preprocessing:
  scaler: standard
  impute_strategy: median

model_params:
  n_estimators: 200
  max_depth: 6
  learning_rate: 0.1

evaluation:
  metrics: [auc, accuracy, significance_improvement]
  save_plots: true
```

## Definition of Done

- [ ] Single command runs full experiment: `python scripts/run_experiment.py --dataset higgs`
- [ ] Results logged to structured JSON/YAML
- [ ] Hyperparameter tuning improves AUC by >2% over defaults
- [ ] Reproducible results with random seed

## Technical Notes

- Use `sklearn.model_selection.GridSearchCV` or `RandomizedSearchCV` for tuning
- Log experiment metadata: timestamp, config hash, git commit
- Consider MLflow for experiment tracking (optional enhancement)
